packages/@ama-sdk/core/src/clients/api-angular-client.ts

```diff
@@ -0,0 +1,160 @@
+  public prepareUrl(url: string, queryParameters: { [key: string]: string | undefined } = {}) {
+    return prepareUrl(url, queryParameters);
+  }
+
+  /** @inheritdoc */
+  public processFormData(data: any, type: string) {
+    return processFormData(data, type);
+  }
+
+  /** @inheritdoc */
+  public tokenizeRequestOptions(url: string, queryParameters: { [key: string]: string }, piiParamTokens: { [key: string]: string }, data: any): TokenizedOptions | undefined {
+    return this.options.enableTokenization ? tokenizeRequestOptions(url, queryParameters, piiParamTokens, data) : undefined;
+  }
+
+  /** @inheritdoc */
+  public async processCall(url: string, options: RequestOptions, apiType: ApiTypes | string, apiName: string, revivers?: undefined, operationId?: string): Promise<void>;
+  public async processCall<T>(url: string, options: RequestOptions, apiType: ApiTypes, apiName: string, revivers: ReviverType<T> | { [statusCode: number]: ReviverType<T> | undefined },
+    operationId?: string): Promise<T>;
+  public async processCall<T>(url: string, options: RequestOptions, apiType: ApiTypes | string, apiName: string,
+    revivers?: ReviverType<T> | undefined | { [statusCode: number]: ReviverType<T> | undefined }, operationId?: string): Promise<T> {
+
+    let response: HttpResponse<any> | undefined;
+    let root: any | undefined;
+    let exception: Error | undefined;
+
+    const origin = options.headers.get('Origin');
+
+    // Execute call
+    try {
+      const headers = Object.fromEntries(options.headers.entries());
+
+      const controller = typeof AbortController !== 'undefined' ? new AbortController() : undefined;
+      if (controller) {
+        options.signal = controller.signal;
+      }
+      const asyncResponse = new Promise<HttpResponse<any>>((resolve, reject) => {
+        let data: HttpResponse<any>;
+        this.options.httpClient.request(options.method, url, {
+          ...options,
+          observe: 'response',
+          headers
+        }).subscribe({
+          next: (res) => data = res,
+          error: (err) => reject(err),
+          complete: () => resolve(data)
+        });
+      });
+      response = await asyncResponse;
+      root = response.body;
+    } catch (e: any) {
+      exception = new EmptyResponseError(e.message || 'Fail to Fetch', undefined, { apiName, operationId, url, origin });
+    }
+
+    // eslint-disable-next-line no-console
+    const reviver = getResponseReviver(revivers, response, operationId, { disableFallback: this.options.disableFallback, log: console.error });
+    const replyPlugins = this.options.replyPlugins ?
+      this.options.replyPlugins.map((plugin) => plugin.load<T>({
+        dictionaries: root && root.dictionaries,
+        response: response && {
+          ...response,
+          headers: new Headers(
+            Array.from(response.headers.keys()
+              .map((key) => ([key, response!.headers.get(key)!] as [string, string]))
+            )
+          )
```

Comment:
isn't it equivalent to:
```suggestion
          headers: new Headers(Array.from(response.headers.entries()))
```
?

Reply:
Actually no because `response.headers.entries()` does not exist :S